### PR TITLE
Change schema.yaml to v2

### PR DESCRIPTION
**Pull request by IronPan**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2009)

### CODE REVIEW

File: manifests/gcp_marketplace/deployer/Dockerfile [hunk 1]
```diff
@@ -1 +1 @@
-FROM gcr.io/cloud-marketplace-tools/k8s/deployer_helm/onbuild
+FROM gcr.io/yang-experiment-6/marketplace-k8s-app-tools/k8s/deployer_helm/onbuild:unreleased-pr396
```
Comment (Ark-kun) [inline manifests/gcp_marketplace/deployer/Dockerfile]: Is this image change temporary?

Comment (IronPan) [reply manifests/gcp_marketplace/deployer/Dockerfile]: updated to permanent gcr

File: manifests/gcp_marketplace/deployer/Dockerfile [hunk 2]
```diff
@@ -1 +1 @@
-FROM gcr.io/cloud-marketplace-tools/k8s/deployer_helm/onbuild
+FROM gcr.io/ml-pipeline/marketplace-k8s-app-tools/k8s/deployer_helm/onbuild:0829-pr396
```
Comment (rileyjbauer) [inline manifests/gcp_marketplace/deployer/Dockerfile]: what's with this tag? is it temporary?

File: manifests/gcp_marketplace/schema.yaml [hunk 3]
```diff
@@ -9,71 +87,26 @@ properties:
...
+    title: CloudSQL zone
+  managedstorage.dbPassword:
+    type: string
+    title: database password
+  managedstorage.gcpProject:
+    type: string
+    title: GCP project
+  managedstorage.databaseNamePrefix:
+    type: string
+    title: databaes name prefix
```
Comment (rileyjbauer) [inline manifests/gcp_marketplace/schema.yaml]: nit: database

File: manifests/gcp_marketplace/chart/kubeflow-pipelines/values.yaml [hunk 4]
```diff
@@ -29,5 +29,5 @@ managedstorage:
   cloudsqlZone: null
   dbPassword: null
   gcpProject: null
-  gcpSecretName: null
   databaseNamePrefix: null
+  gcpSecretName: null
```
Comment (james-jwu) [inline manifests/gcp_marketplace/chart/kubeflow-pipelines/values.yaml]: Please sort

File: manifests/gcp_marketplace/deployer/Dockerfile (hunk 2 above)
Comment (IronPan) [reply manifests/gcp_marketplace/deployer/Dockerfile]: yes this is an unofficial image. the change is not in marketplace prod yet.

Comment (IronPan) [reply manifests/gcp_marketplace/deployer/Dockerfile]: i talked with marketplace and using this should be fine for now.

File: manifests/gcp_marketplace/chart/kubeflow-pipelines/values.yaml (hunk 4 above)
Comment (IronPan) [reply manifests/gcp_marketplace/chart/kubeflow-pipelines/values.yaml]: done

File: manifests/gcp_marketplace/schema.yaml (hunk 3 above)
Comment (IronPan) [reply manifests/gcp_marketplace/schema.yaml]: done

**Pull request merged**